### PR TITLE
refactor(tracer): separate dogstatsd for AgentWriter and RuntimeWorker

### PR DESCRIPTION
**Pull request by majorgreys**
The tracer initializes a single dogstatsd client and shares it between the two worker threads. This would better be handled separately. The flush interval for the runtime metrics worker is also decoupled from the tracer and exposed as an environment variable.

### CODE REVIEW

File: ddtrace/tracer.py [hunk 1]
```diff
@@ -118,15 +91,15 @@ def __init__(self, url=None, dogstatsd_url=DEFAULT_DOGSTATSD_URL):
 
         uds_path = None
         https = None
-        hostname = self.DEFAULT_HOSTNAME
-        port = self.DEFAULT_PORT
+        hostname = environ.get("DD_AGENT_HOST", environ.get("DATADOG_TRACE_AGENT_HOSTNAME")) or self.DEFAULT_HOSTNAME
```
Comment (Kyle-Verhoog) [inline ddtrace/tracer.py:94]: the hostname/port changes are unrelated to this patch and so they can be moved to #1848 or a separate PR

File: ddtrace/internal/utils.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from ddtrace.compat import parse
+from ddtrace.vendor.dogstatsd import DogStatsd
+
+
+def _parse_dogstatsd_url(url):
```
Comment (Kyle-Verhoog) [inline ddtrace/internal/utils.py:5]: I wonder if this should go in utils/formats 🤔

File: ddtrace/tracer.py (hunk 1 above)
Comment (majorgreys) [reply ddtrace/tracer.py]: 👍🏽

File: ddtrace/internal/utils.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+from ddtrace.compat import parse
```
[the remark this answers is not in the document]
Comment (majorgreys) [reply ddtrace/internal/utils.py]: 👍🏽

File: ddtrace/internal/utils.py (hunk 2 above)
Comment (majorgreys) [reply ddtrace/internal/utils.py]: I kept this together with the dogstatsd connection handling as it doesn't seem generally applicable. I could see a benefit from refactoring the url parsing for the trace and metrics.

File: ddtrace/tracer.py [hunk 4]
```diff
@@ -525,22 +499,16 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
...
-        # DEV: ddstatsd expects tags in the form ['key1:value1', 'key2:value2', ...]
-        tags = ["{}:{}".format(k, v) for k, v in RuntimeTags()]
-        self.log.debug("Updating constant tags %s", tags)
-        self._dogstatsd_client.constant_tags = tags
-
     def _start_runtime_worker(self):
-        self._runtime_worker = RuntimeWorker(self._dogstatsd_client, self._RUNTIME_METRICS_INTERVAL)
-        self._runtime_worker.start()
+        self._runtime_worker = RuntimeWorker(self._dogstatsd_url)
+        self._runtime_metrics_started = True
```
Comment (Kyle-Verhoog) [inline ddtrace/tracer.py:511]: This doesn't seem to be used or required since we can use `self._runtime_worker.started`

File: ddtrace/tracer.py [hunk 5]
```diff
@@ -575,10 +543,9 @@ def _check_new_process(self):
 
         if self._runtime_worker is not None:
             self._start_runtime_worker()
-
-        # force an immediate update constant tags since we have reset services
-        # and generated a new runtime id
-        self._update_dogstatsd_constant_tags()
+            # force an immediate update constant tags since we have reset services
+            # and generated a new runtime id
+            self._runtime_worker.update_runtime_tags()
```
Comment (Kyle-Verhoog) [inline ddtrace/tracer.py:548]: Does it make sense to do this in `_start_runtime_worker` instead? Seems like we'd always want to update the tags when creating a new runtime worker

File: ddtrace/tracer.py (hunk 4 above)
Comment (majorgreys) [reply ddtrace/tracer.py:511]: Removed, this is relevant to #1992 .

File: ddtrace/tracer.py (hunk 5 above)
Comment (majorgreys) [reply ddtrace/tracer.py]: 👍🏽

File: ddtrace/tracer.py [hunk 6]
```diff
@@ -295,17 +271,16 @@ def configure(
...
+
         if dogstatsd_host is not None and dogstatsd_url is None:
-            dogstatsd_url = "udp://{}:{}".format(dogstatsd_host, dogstatsd_port or self.DEFAULT_DOGSTATSD_PORT)
+            dogstatsd_url = "udp://{}:{}".format(dogstatsd_host, dogstatsd_port)
 
-        if dogstatsd_url is not None:
-            dogstatsd_kwargs = _parse_dogstatsd_url(dogstatsd_url)
-            self.log.debug("Connecting to DogStatsd(%s)", dogstatsd_url)
-            self._dogstatsd_client = DogStatsd(**dogstatsd_kwargs)
+        self._dogstatsd_url = dogstatsd_url or get_env("dogstatsd", "url", default=self.DEFAULT_DOGSTATSD_URL)
```
Comment (Kyle-Verhoog) [inline ddtrace/tracer.py:279]: Just need to update these once #2044 is merged

**Review by Kyle-Verhoog — COMMENTED** [review]
Looks good to me!

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]
